manager/src/main/java/org/openremote/manager/mqtt/AssetInterceptHandler.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2017, OpenRemote Inc.
+ *
+ * See the CONTRIBUTORS.txt file in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.openremote.manager.mqtt;
+
+import io.moquette.interception.InterceptHandler;
+import io.moquette.interception.messages.*;
+import org.keycloak.adapters.rotation.AdapterTokenVerifier;
+import org.keycloak.common.VerificationException;
+import org.keycloak.representations.AccessToken;
+import org.openremote.container.message.MessageBrokerService;
+import org.openremote.container.security.ClientCredentialsAuthFrom;
+import org.openremote.container.security.keycloak.AccessTokenAuthContext;
+import org.openremote.container.util.UniqueIdentifierGenerator;
+import org.openremote.container.web.socket.WebsocketConstants;
+import org.openremote.manager.asset.AssetProcessingService;
+import org.openremote.manager.asset.AssetStorageService;
+import org.openremote.manager.event.ClientEventService;
+import org.openremote.manager.security.ManagerIdentityService;
+import org.openremote.manager.security.ManagerKeycloakIdentityProvider;
+import org.openremote.model.Constants;
+import org.openremote.model.asset.AssetFilter;
+import org.openremote.model.attribute.AttributeEvent;
+import org.openremote.model.attribute.AttributeRef;
+import org.openremote.model.event.shared.CancelEventSubscription;
+import org.openremote.model.event.shared.EventSubscription;
+import org.openremote.model.event.shared.RenewEventSubscriptions;
+import org.openremote.model.interop.BiConsumer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import static org.openremote.manager.mqtt.MqttBrokerService.TOPIC_SEPARATOR;
+import static org.openremote.model.Constants.KEYCLOAK_CLIENT_ID;
+
+public class AssetInterceptHandler implements InterceptHandler {
```

Comment:
Maybe give this a more generic name rather than `AssetInterceptHandler` as we could allow other `SharedEvent`s to be pubbed/subbed over `MQQT`